### PR TITLE
[Chuong] fix UI

### PR DESCRIPTION
**Pull request by onstonboy**
done

### CODE REVIEW

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumPresenter.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Listens to user actions from the UI ({@link SongBelongAlbumActivity}), retrieves the data and
+ * updates
+ * the UI as required.
+ */
+final class SongBelongAlbumPresenter implements SongBelongAlbumContract.Presenter {
+    private static final String TAG = SongBelongAlbumPresenter.class.getName();
+
+    private SongBelongAlbumContract.View mView;
+
+    public SongBelongAlbumPresenter() {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumPresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistPresenter.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * Listens to user actions from the UI ({@link SongBelongArtistActivity}), retrieves the data and
+ * updates
+ * the UI as required.
+ */
+final class SongBelongArtistPresenter implements SongBelongArtistContract.Presenter {
+    private static final String TAG = SongBelongArtistPresenter.class.getName();
+
+    private SongBelongArtistContract.View mView;
+
+    public SongBelongArtistPresenter() {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistPresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumContract.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import com.fstyle.androidtrainning.screen.BaseView;
+
+/**
+ * This specifies the contract between the view and the presenter.
+ */
+interface SongBelongAlbumContract {
+    /**
+     * View.
+     */
+    interface View extends BaseView {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumContract.java]: WARNING: [ShortClassName] Avoid short class names like View
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistContract.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import com.fstyle.androidtrainning.screen.BaseView;
+
+/**
+ * This specifies the contract between the view and the presenter.
+ */
+interface SongBelongArtistContract {
+    /**
+     * View.
+     */
+    interface View extends BaseView {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistContract.java]: WARNING: [ShortClassName] Avoid short class names like View
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    public SongBelongArtistRecyclerAdapter(Context context) {
+        mContext = context;
+    }
+
+    public void updateData(List<Track> tracks) {
+        if (tracks == null) {
+            return;
+        }
+        mTracks = tracks;
+        notifyDataSetChanged();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java:33]: ERROR: [PMD-CPD] 58 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    @Override
+    public int getItemCount() {
+        return mTracks.size();
+    }
+
+    public class RecyclerViewHolder extends RecyclerView.ViewHolder {
+        private TextView mTxtNameSong, mTxtNameSinger;
+        private static final int MEDIUM_IMAGE = 1;
+        private static final int MAX_LENGTH = 25;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java:56]: ERROR: [PMD-CPD] 36 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java
app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @Override
+    public int getItemCount() {
+        return mTracks.size();
+    }
+
+    public class RecyclerViewHolder extends RecyclerView.ViewHolder {
+        private TextView mTxtNameSong, mTxtNameSinger;
+        private static final int MAX_LENGTH = 25;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java:55]: ERROR: [PMD-CPD] 36 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java
app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/data/local/storage/ExternalData.java [hunk 8]
```diff
@@ -183,7 +201,7 @@ public void scanAllMusicBelongArtist(Context context, String temp) {
                 MediaStore.Audio.Media.ALBUM_ID, MediaStore.Audio.Media.ARTIST,
                 MediaStore.Audio.Media.DURATION, MediaStore.Audio.Media.DATA
         };
-        String where = MediaStore.Audio.Media.ALBUM + LIKE;
+        String where = MediaStore.Audio.Media.ARTIST + LIKE;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/data/local/storage/ExternalData.java:204]: ERROR: [PMD-CPD] 23 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/data/local/storage/ExternalData.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+ * Created by Administrator on 12/15/17.
+ */
+
+public class SongBelongAlbumRecyclerAdapter
+        extends RecyclerView.Adapter<SongBelongAlbumRecyclerAdapter.RecyclerViewHolder> {
+
+    private List<Track> mTracks = new ArrayList<>();
+    private Context mContext;
+
+    public SongBelongAlbumRecyclerAdapter(Context context) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java:24]: ERROR: [PMD-CPD] 32 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+ * Created by Administrator on 12/15/17.
+ */
+
+public class SongBelongArtistRecyclerAdapter
+        extends RecyclerView.Adapter<SongBelongArtistRecyclerAdapter.RecyclerViewHolder> {
+
+    private List<Track> mTracks = new ArrayList<>();
+    private Context mContext;
+
+    public SongBelongArtistRecyclerAdapter(Context context) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java:24]: ERROR: [PMD-CPD] 32 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+            mTxtNameSinger = itemView.findViewById(R.id.text_lower);
+        }
+
+        public void bind(int position) {
+            setNameArtist(position);
+            setNameSong(position);
+        }
+
+        private void setNameArtist(int position) {
+            String nameArtist = mTracks.get(position).getNameArtist();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java:72]: ERROR: [PMD-CPD] 20 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java
app/src/main/java/com/fstyle/androidtrainning/screen/searchonline/SearchSongRecyclerAdapter.java
app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+            mTxtNameSinger = itemView.findViewById(R.id.text_lower);
+        }
+
+        public void bind(int position) {
+            setNameArtist(position);
+            setNameSong(position);
+        }
+
+        private void setNameArtist(int position) {
+            String nameArtist = mTracks.get(position).getNameArtist();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongartist/SongBelongArtistRecyclerAdapter.java:71]: ERROR: [PMD-CPD] 20 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java
app/src/main/java/com/fstyle/androidtrainning/screen/searchonline/SearchSongRecyclerAdapter.java
app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+    public SongBelongAlbumRecyclerAdapter(Context context) {
+        mContext = context;
+    }
+
+    public void updateData(List<Track> tracks) {
+        if (tracks == null) {
+            return;
+        }
+        mTracks = tracks;
+        notifyDataSetChanged();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumRecyclerAdapter.java:33]: ERROR: [PMD-CPD] 23 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/searchoffline/SearchSongRecyclerAdapter.java

File: app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumActivity.java [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+import com.fstyle.androidtrainning.R;
+import com.fstyle.androidtrainning.data.local.storage.ExternalData;
+import com.fstyle.androidtrainning.model.Album;
+import com.fstyle.androidtrainning.screen.BaseActivity;
+import com.fstyle.androidtrainning.utils.Constant;
+
+/**
+ * Songbelongalbum Screen.
+ */
+public class SongBelongAlbumActivity extends BaseActivity implements SongBelongAlbumContract.Viewer {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/songbelongalbum/SongBelongAlbumActivity.java]: ERROR: Line is longer than 100 characters (found 101).

**Comment by tranminhtri911** [comment]
ok

**Pull request merged**